nuxeo-core/nuxeo-core-io/src/main/java/org/nuxeo/ecm/core/io/marshallers/json/enrichers/AbstractJsonEnricher.java

```diff
@@ -51,8 +55,18 @@ public boolean accept(Class<?> clazz, Type genericType, MediaType mediatype) {
     }
 
     @Override
-    public void write(Enriched<EntityType> enrichable, JsonGenerator jg) throws IOException {
-        write(jg, enrichable.getEntity());
+    public void write(Enriched<EntityType> enrichable, JsonGenerator jg) {
+        try {
+            write(jg, enrichable.getEntity());
+        } catch (Exception e) {
+            if (e instanceof InterruptedException) {
+                Thread.currentThread().interrupt();
+                throw new RuntimeException("interrupted", e); // NOSONAR
+            } else {
+                // TODO collect exception and return it to the caller
+                log.error(e, e);
```

Comment:
Thinking of it, I would log it at `INFO` or `DEBUG` level. Not sure it's a good idea to fill the server logs with errors when doing calls asking for an enricher that fails.
Looking at the initial issue [NXP-28047](https://jira.nuxeo.com/browse/NXP-28047), it seems legitimate to me to define a set of enrichers used for all calls of an application, even if they should not be applicable for all calls.

WDYT @kevinleturc @efge ?

Reply:
But even then these enrichers shouldn't fail with an exception, they should gracefully return nothing. I really don't see a good reason to hide an exception here.

Reply:
I'm in favor of an info log with a specific message, not the exception, and keep the exception as 2nd argument for the stack.

Also, we can have something like this (but overkill IMHO):
```java
if (log.isInfoEnabled()) {
    log.info("message", e);
} else {
    log.warn("message");
}
```

Reply:
Since I can't make sure how each writer does handle  failure and some may return exceptions when they should just fail gracefully, log.info seems to be a workaround to make them fail silently if not gracefully.

About the conditional structure, I don't think it is overkill. Because in the case info is not enabled then people might search why enrichment is not done in the requests rather than in the writer. But I went for a simple `log.info("enrichment failed", e);`

Reply:
There's still not the message in the code 🤔did you push?

Reply:
I did, but that would override the message in the exception thrown by the writer.
I found the message thrown by FailingEnricher to be more specific than a static one I would put here
So I reverted it when complying to https://github.com/nuxeo/nuxeo/pull/3706#discussion_r372956789